### ISSUE TITLE
FATAL: module compiled as little endian, but detected different endianness at runtime

### ISSUE THREAD

**Issue by jkotan** [pni-libraries/python-pni#73]
I've just tried to run tests on debian buster with python3. It looks like it fails with

    Error: FATAL: module compiled as little endian, but detected different endianness at runtime

because of python-boost (1.67.0) problem: https://github.com/boostorg/python/pull/218

It should be fixed in 1.68.0

**Comment by jkotan** [comment]
Tests are at: https://travis-ci.org/pni-libraries/python-pni/jobs/458359525

**Comment by eugenwintersberger** [comment]
This is quite surprising as the problem appears on my buster installation even with python-boost 1.62.0. Is this the same issue?

**Comment by jkotan** [comment]
https://github.com/boostorg/python/issues/209

**Comment by jkotan** [comment]
Probably, the problem comes when you compile with g++8

**Comment by eugenwintersberger** [comment]
We definitely should try to get rid of Boot Python. This thing is a nightmare. I am currently evaluating pybind11. It is header only. But I do not know how well it works on Debian 9.

**Comment by jkotan** [comment]
It sounds nice. I hope it works al well ;)

**Comment by jkotan** [comment]
It looks like it was fixed in boost  [tests](https://travis-ci.org/pni-libraries/python-pni/jobs/470908749). However, If we want to compile with 1.67 or higher we need a patch:
```patch
diff --git a/setup.py b/setup.py
index d1ecb6f..b550698 100644
--- a/setup.py
+++ b/setup.py
@@ -56,8 +56,8 @@ else:
     nexus_config.add_link_library('pniio')
     nexus_config.add_link_library('h5cpp')
     nexus_config.add_link_library(
-        "boost_python-py{major}{minor}".format(major=sys.version_info.major,
-                                               minor=sys.version_info.minor))
+        "boost_python{major}{minor}".format(major=sys.version_info.major,
+                                            minor=sys.version_info.minor))
     nexus_config.add_include_directory('/usr/include/hdf5/serial')
 ```
because of `.so` name changes. 
I've figured out how to easily detect the current boost version in setup.py (except parsing header files or compiling small program) so I keep buster branch separately [buster](https://github.com/pni-libraries/python-pni/tree/buster) .

BTW, I've recompile `buster` and `cosmic` `pni` packages and put it into the HDRI repo.

**Comment by jkotan** [comment]
It looks like cosmic still suffer from this bug: https://travis-ci.org/pni-libraries/python-pni/jobs/476777181

**Comment by jkotan** [comment]
On debian buster  we still get error messages, i.e.
```python
In [1]: from pninexus import h5cpp
---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
RuntimeError: FATAL: module compiled as little endian, but detected different endianness at runtime
---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
RuntimeError: FATAL: module compiled as little endian, but detected different endianness at runtime
---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
RuntimeError: FATAL: module compiled as little endian, but detected different endianness at runtime

In [2]: h5cpp
Out[2]: <module 'pninexus.h5cpp' from '/usr/lib/python3/dist-packages/pninexus/h5cpp/__init__.py'>
```
but `h5cpp` works.

**Comment by jkotan** [comment]
Since the fix from  `buster` with apply to other new distributions I've renamed `buster` branch to `develop`

**Comment by jkotan** [comment]
fixed in 1.2.11

**Comment by ming82871** [comment]
> Probably, the problem comes when you compile with g++8

I fixed the problem when compiled with g++7
